apps/studio/components/interfaces/Settings/General/ComplianceConfig/ProjectComplianceMode.tsx

```diff
@@ -0,0 +1,122 @@
+import { PermissionAction } from '@supabase/shared-types/out/constants'
+import { Loader2 } from 'lucide-react'
+import { useEffect, useState } from 'react'
+import { toast } from 'sonner'
+
+import { useParams } from 'common'
+import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
+import { DocsButton } from 'components/ui/DocsButton'
+import { FormHeader } from 'components/ui/Forms/FormHeader'
+import { FormPanel } from 'components/ui/Forms/FormPanel'
+import { FormSection, FormSectionContent, FormSectionLabel } from 'components/ui/Forms/FormSection'
+import { useProjectSettingsV2Query } from 'data/config/project-settings-v2-query'
+import { useComplianceConfigUpdateMutation } from 'data/config/project-compliance-config-mutation'
+import { useCheckPermissions } from 'hooks/misc/useCheckPermissions'
+import { subscriptionHasHipaaAddon } from 'components/interfaces/Billing/Subscription/Subscription.utils'
+import { useOrgSubscriptionQuery } from 'data/subscriptions/org-subscription-query'
+import { useSelectedOrganization } from 'hooks/misc/useSelectedOrganization'
+
+import { Switch, TooltipContent_Shadcn_, TooltipTrigger_Shadcn_, Tooltip_Shadcn_ } from 'ui'
+
+const ComplianceConfig = () => {
+  const { ref } = useParams()
+  const [isSensitive, setIsSensitive] = useState(false)
+  const selectedOrganization = useSelectedOrganization()
+  const { data: subscription } = useOrgSubscriptionQuery({ orgSlug: selectedOrganization?.slug })
+  const hasHipaaAddon = subscriptionHasHipaaAddon(subscription)
+  const { data: settings, isLoading, isSuccess } = useProjectSettingsV2Query({ projectRef: ref })
```

Comment:
```suggestion
  const { data: settings, isLoading, isSuccess } = useProjectSettingsV2Query({ projectRef: ref }, { enabled: hasHipaaAddon })
```

Would prevent the request from being triggered unless customer actually has the HIPAA add-on